src/core/thread/mle_router.cpp

```diff
@@ -1540,8 +1565,9 @@ void MleRouter::HandleTimeTick(void)
     switch (mRole)
     {
     case kRoleDetached:
-        if (mChallengeTimeout == 0)
+        if (mChallengeTimeout == 0 && mLinkRequestAttempts == mMaxLinkRequests)
         {
+            mInRouterResetSync = false; // Router has retransmitted link request maximum times, give up sync process
```

Comment:
Follow up on previous comment on `mChallengeTimeout`.

- With this PR we now do multiple `SendLinkRequest(nullptr)` before giving up (if no response) and get here and try to attach as child.
- We wait a random interval `[4.5-5.5]` second between link request attempts (average of 4 seconds).
- From  `SendLinkRequest(nullptr)`, `mChallengeTimeout` is set to `2`.
- In `HandleLinkAccept()` we require that `(mChallengeTimeout > 0)` to accept the message
- So we wait for `~4` seconds in between retx but accept response only for first `~2` seconds.

I feel may be nicer to accept response while we still have remaining `mLinkRequestAttempts` and waiting for next retx.

Thoughts?

Reply:
I agree it's nice to
* accept response while we still have remaining `mLinkRequestAttempts`.
* only accept response for a reasonable time after we send link request (and not going to resend)

So I set `mLinkRequestAttempts` to 0 when we have got response or we give up the sync, and updated `HandleLinkAccept()` to accept the message if `mLinkRequestAttempts > 0` or `mChallengeTimeout > 0`, I think this should achieve what we want.

Reply:
Yes. This should work and is simple. Thanks.